src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -541,8 +542,7 @@ private RecoveryAwareAMQConnection recoverConnection() throws InterruptedExcepti
 				newConn.abort();
 				return null;
             } catch (Exception e) {
-                // TODO: exponential back-off
-                Thread.sleep(this.params.getNetworkRecoveryInterval());
+                Thread.sleep(this.params.getRecoveryDelayHandler().getDelay(attempts));
```

Comment:
thoughts on flipping the order of this? so deliver the exception before sleeping?

Reply:
Let's keep it the way it is.

Reply:
Good catch. I think at some point we will have to do it. The way it works right now is a bit weird and can complicate additional manual recovery steps.